pkg/registry/core/pod/storage/eviction.go

```diff
@@ -34,7 +35,9 @@ import (
 	policyclient "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
 	"k8s.io/client-go/util/retry"
 	api "k8s.io/kubernetes/pkg/apis/core"
+	corev1 "k8s.io/kubernetes/pkg/apis/core/v1"
 	"k8s.io/kubernetes/pkg/apis/policy"
+	"k8s.io/kubernetes/pkg/controller/disruption"
```

Comment:
API should not take dependencies on controllers, it should be to a third party package.  The method you carved off should be in podutil, not in controller, and both places should reference it.

Reply:
Also, I'm not really a fan of sharing a wrapper when the underlying condition is more clear.  We don't have a "Healthy" concept in the API.  We have "Ready".  Both controller and API should be defined in terms of "Ready", or have local conditions.  But creating a synthetic API "healthy" in shared code is something I generally oppose because it is not modelled as part of our API.

Reply:
I'll just remove that change, podutil.IsPodReady(pod) is already being called, so I'll just use that in the eviction api.

Reply:
I agree with not using the terminology 'healthy' but that's what PDBs expose: https://godoc.org/k8s.io/api/policy/v1beta1#PodDisruptionBudgetStatus

I can't find any specific wording to indicate that Healthy == Ready.